src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ReplayDataPlaybackController.cs

```diff
@@ -79,25 +79,28 @@ void OnSceneLoad(Scene s, LoadSceneMode m)
 
         private void SetupEventSystem()
         {
-            // when/if we can make the legacy input system work, we should remove this and respect that system
             var eventSystems = FindObjectsByType<EventSystem>(FindObjectsSortMode.None);
-            InputSystemUIInputModule inputModule = null;
             foreach (var eventSystem in eventSystems)
             {
-                var theModule = eventSystem.gameObject.GetComponent<InputSystemUIInputModule>();
-                if (theModule != null)
+                BaseInputModule inputModule = eventSystem.gameObject.GetComponent<BaseInputModule>();
+                
+                // If there is no module, add the appropriate input module so that the replay can simulate UI inputs
```

Comment:
Updated the comment with the clarification that, if the input module is not already present, it will prefer to use the new UI input system module over the old one.